coalib/bears/Bear.py

```diff
@@ -305,6 +316,33 @@ def log_message(self, log_message, timestamp=None, **kwargs):
     def run(self, *args, dependency_results=None, **kwargs):
         raise NotImplementedError
 
+    def _dump_bear_profiled_file(self, profiler):
+        filename = '{}_{}.prof'.format(self.section.name, self.name)
+        if self.profile is True:
+            profiler.dump_stats(filename)
+        else:
+            if not (os.path.exists(self.profile) and
+                    os.path.isdir(self.profile)):
```

Comment:
is `exists` necessary if you check with `isdir`?

Reply:
NO :+1:  `isdir` first check it exits :+1: